src/PIL/PyAccess.py

```diff
@@ -357,9 +365,13 @@ def set_pixel(self, x, y, color):
     mode_map["I;32B"] = _PyAccessI32_N
 
 
-def new(img, readonly=False):
+def new(img: Image.Image, readonly: bool = False) -> PyAccess | None:
     access_type = mode_map.get(img.mode, None)
     if not access_type:
         logger.debug("PyAccess Not Implemented: %s", img.mode)
         return None
     return access_type(img, readonly)
+
+
+if TYPE_CHECKING:
+    from . import Image
```

Comment:
You didn't want to put this import at the top of the file?

Reply:
For lighter/quicker imports when not type checking, and potentially avoid circular imports?

Reply:
The convention so far has been to put the `TYPE_CHECKING` imports just before their use - [`OPEN`](https://github.com/python-pillow/Pillow/blob/58a47978af9f34851ce926303d05fa677010ce2a/src/PIL/Image.py#L220-L221), [`Exif`](https://github.com/python-pillow/Pillow/blob/58a47978af9f34851ce926303d05fa677010ce2a/src/PIL/Image.py#L3698-L3701), [`ImageFileDirectory_v2`](https://github.com/python-pillow/Pillow/blob/58a47978af9f34851ce926303d05fa677010ce2a/src/PIL/TiffImagePlugin.py#L484-L487) and [`PdfDict`](https://github.com/python-pillow/Pillow/blob/58a47978af9f34851ce926303d05fa677010ce2a/src/PIL/PdfParser.py#L251-L254)

Reply:
> For lighter/quicker imports when not type checking, and potentially avoid circular imports?

I hadn't realized mypy has no problem dealing with circular imports. I've now moved the import near the top (still with the `TYPE_CHECKING` check).